projects/storefrontlib/src/lib/my-account/update-profile/components/update-profile-form.component.spec.ts

```diff
@@ -0,0 +1,167 @@
+import { DebugElement } from '@angular/core';
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { ReactiveFormsModule } from '@angular/forms';
+import { By } from '@angular/platform-browser';
+import { User } from '@spartacus/core';
+import { UpdateProfileFormComponent } from './update-profile-form.component';
+
+const mockUser: User = {
+  titleCode: 'dr',
+  firstName: 'N',
+  lastName: 'Z',
+  uid: 'xxx@xxx.xxx',
+};
+
+describe('UpdateProfileFormComponent', () => {
+  let component: UpdateProfileFormComponent;
+  let fixture: ComponentFixture<UpdateProfileFormComponent>;
+  let el: DebugElement;
+  let form: DebugElement;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [ReactiveFormsModule],
+      declarations: [UpdateProfileFormComponent],
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(UpdateProfileFormComponent);
+    component = fixture.componentInstance;
+    el = fixture.debugElement;
+    form = fixture.debugElement.query(By.css('form'));
+
+    fixture.detectChanges();
+    component.user = mockUser;
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  describe('isNotValid', () => {
+    it('should return true if NOT valid', () => {
+      component.ngOnInit();
+
+      expect(component.isNotValid('firstName')).toEqual(false);
+    });
+
+    it(`should return false if the form was not submitted even though it's invalid`, () => {
+      const invalidUser: User = {
+        ...mockUser,
+        firstName: '',
+      };
+      component.user = invalidUser;
+      component['submitClicked'] = false;
+
+      component.ngOnInit();
+      component.form.get('firstName').markAsDirty();
+
+      expect(component.isNotValid('firstName')).toEqual(false);
+    });
+
+    it('should return false if valid', () => {
+      const invalidUser: User = {
+        ...mockUser,
+        firstName: '',
+      };
+      component.user = invalidUser;
+      component['submitClicked'] = true;
+
+      component.ngOnInit();
+      component.form.get('firstName').markAsDirty();
+
+      expect(component.isNotValid('firstName')).toEqual(true);
+    });
+  });
+
+  describe('onSubmit', () => {
+    it('should call onSubmit() on submit event', () => {
+      spyOn(component, 'onSubmit').and.stub();
+      form.triggerEventHandler('ngSubmit', null);
```

Comment:
Comment for the submit tests: would clicking on the button via the dom element instead cover more possible issues, like the button not being displayed or being disabled.

Reply:
done